vault/resource_mongodbatlas_secret_backend_test.go

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package vault
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+	"github.com/hashicorp/terraform-provider-vault/testutil"
+)
+
+func TestAccMongoDBAtlasSecretBackend_basic(t *testing.T) {
+	mount := acctest.RandomWithPrefix("tf-test-mongodbatlas")
+	resourceType := "vault_mongodbatlas_secret_backend"
+	resourceName := resourceType + ".test"
+	privateKey, publicKey := testutil.GetTestMDBACreds(t)
+
+	updatedPrivateKey := "905ae89e-6ee8-40rd-ab12-613t8e3fe836"
+	updatedPublicKey := "klpruxce"
+
+	resource.Test(t, resource.TestCase{
+		ProviderFactories: providerFactories,
+		PreCheck:          func() { testutil.TestAccPreCheck(t) },
+		CheckDestroy:      testCheckMountDestroyed(resourceType, consts.MountTypeMongoDBAtlas, consts.FieldPath),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccMongoDBAtlasSecretBackendConfig_basic(mount, privateKey, publicKey),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, consts.FieldBackend, mount),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldPath, fmt.Sprintf("%s/config", mount)),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldPrivateKey, privateKey),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldPublicKey, publicKey),
+				),
+			},
+			{
+				Config: testAccMongoDBAtlasSecretBackendConfig_basic(mount, updatedPrivateKey, updatedPublicKey),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, consts.FieldBackend, mount),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldPath, fmt.Sprintf("%s/config", mount)),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldPrivateKey, updatedPrivateKey),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldPublicKey, updatedPublicKey),
+				),
+			},
+			{
+				ResourceName:            resourceName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{consts.FieldDisableRemount, consts.FieldPrivateKey},
```

Comment:
Can remove this as per comment above; this field no longer exists in the schema
```suggestion
				ImportStateVerifyIgnore: []string{consts.FieldPrivateKey},
```